### ISSUE TITLE
Enable className on TextInput component

### ISSUE THREAD

**Issue by jdelight** [BrandwatchLtd/axiom-react#919]
Not 100% sure where it should end up - perhaps the input container (i.e. wherever space classes end up).

**Comment by lpoulter** [comment]
There are a couple of different options off where to put the classname.
- The input wrapper
- The input element

I'd guess that 1 is the best place as often Axiom users want to adjust border box or positional CSS properties. What was your use-case?

**Comment by lpoulter** [comment]
Hey @jdelight do you remember what your use case was for the classname to help decide where best to place it?

**Comment by jdelight** [comment]
Hey @lpoulter I can't remember the specific use case now however I think we should be aiming for consistency across all components / composites rather than addressing specific use cases as Axiom needs to support multiple use cases. 

Generally with simple components it's easy to know where a className prop should end up. However, in this case (and there will be others like this) - we have multiple elements being composed together.

If a developer was creating this construct without Axiom, they would be able to apply classNames anywhere so I think we should extend the same idea to Axiom components which compose multiple HTML elements.

Two ideas come to mind:

1. use a BEM-like convention which most developers are used to:

In this solution a dev passes in a className prop which is used as a prefix on the relevant sub elements:

```jsx
<TextInput className="myclass" / >
```
Would render something like the following:

```html
<label class="myclass__textinput-container"><input class="myclass__textinput-input" /></label>
```
The benefit here is that the external API of just set a `className` prop is preserved but the developer looses some control over the specific className eventually created and this needs to be documented.

2. Define specific className props for each relevant element

In this solution we can expose some props to the `TextInput` component which will be passed down to the relevant low level HTML element

```jsx
<TextInput inputClassName="myinput" containerClassName="mycontainer" / >
```
Would render something like the following:

```html
<label class="mycontainer"><input class="myinput" /></label>
```

The benefit of this is level of control afforded to the developer, it's also incredibly clear how this works.
Of the two approaches this is one I've seen in a few places (for example, the react datepicker library will expose props like `dayClassName` or `calendarClassName` as a way to provide hooks into the underlying HTML).

For the TextInput I think the two elements you mentioned would be perfect places for className hooks, so perhaps `className` could refer to the container, and `inputClassName` could refer to the input. 

This might lead to a convention whereby whenever we render a complex/nested HTML elements, `className` by default refers to the outermost element with more specific props created for internals where necessary.

What do you think?

**Comment by lpoulter** [comment]
_className by default refers to the outermost element with more specific props created for internals where necessary._
Love it.

**Comment by lpoulter** [comment]
Thanks for your input I'll do this for `TextInput` and see how it feels.

**Comment by lpoulter** [comment]
Closed by #936
